### PR TITLE
Install `cri-tools` from `master`

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind ci

#### What this PR does / why we need it:
Use their corresponding main branches for testing instead of the pinned version.
#### Which issue(s) this PR fixes:

None
#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: .github/workflows/integration.yml [hunk 1]
```diff
@@ -9,7 +9,7 @@ on:
       - update-nixpkgs-*
   pull_request:
 env:
-  GO_VERSION: '1.22'
+  GO_VERSION: '1.21'
```
Comment (sohankunkerkar) [inline .github/workflows/integration.yml:12]: I think you can still keep the Go version as 1.22 and build runc with 1.21.9 (https://github.com/cri-o/cri-o/pull/8011/files#diff-2864e5b0ec49a60b63556bdbf2ffa9c257ebae8656bb9d9ebbbf718d3eb40e8cR111) unless we decided to change the version to 1.21

Comment (haircommander) [reply .github/workflows/integration.yml:12]: I thought runc main branch is failing to build with any go 1.22 right now

Comment (sohankunkerkar) [reply .github/workflows/integration.yml:12]: Yeah, that's why we need to use the same hack as the one used in CI: https://github.com/cri-o/cri-o/pull/8011/files#diff-2864e5b0ec49a60b63556bdbf2ffa9c257ebae8656bb9d9ebbbf718d3eb40e8cR109-R129

Comment (saschagrunert) [reply .github/workflows/integration.yml]: Reverted as suggested.

File: dependencies.yaml [hunk 2]
```diff
@@ -57,12 +55,6 @@ dependencies:
     - path: scripts/versions
       match: buildah
 
-  - name: runc
```
Comment (sohankunkerkar) [inline dependencies.yaml:60]: I would retain this logic (similar to CI). There may be instances where we need to stick to a specific version due to regressions in runc

Comment (saschagrunert) [reply dependencies.yaml]: Alright, reverted that change.

**Review by sohankunkerkar — APPROVED** [review]
/lgtm

**Pull request merged**